src/openforms/prefill/contrib/haalcentraal/plugin.py

```diff
@@ -72,25 +72,35 @@ def _get_values_for_bsn(
 
         return values
 
-    @classmethod
-    def get_prefill_values(
-        cls, submission: Submission, attributes: Sequence[str]
-    ) -> dict[str, Any]:
+    def get_identifier_value(
+        self, submission: Submission, identifier_role: str
+    ) -> str | None:
+        if not submission.is_authenticated:
+            return
+
         if (
-            not submission.is_authenticated
-            or submission.auth_info.attribute != AuthAttribute.bsn
+            identifier_role == IdentifierRoles.main
+            and submission.auth_info.attribute == self.requires_auth
         ):
-            #  If there is no bsn we can't prefill any values so just return
-            logger.info("No BSN associated with submission, cannot prefill.")
+            return submission.auth_info.value
+
+        if identifier_role == IdentifierRoles.authorised_person:
+            return submission.auth_info.machtigen.get("value")
+
+    def get_prefill_values(
+        self,
```

Comment:
I don't really see a reason why this can't be a classmethod still? None of the `self` instance attributes seem to be used.

Reply:
Should turn `get_identifier_value` into a classmethod too, in that case.

Reply:
In `get_identifier_value` we need `self.requires_auth` :S

Reply:
but that's a class attribute so should be available through `cls.requires_auth`, no?

Reply:
Oh yes, it's a class attribute... I'm a silly :facepalm: